client/app/lib/utils.js

```diff
@@ -166,3 +167,16 @@ export function formatColumnValue(value, columnType = null) {
 
   return value;
 }
+
+/**
+ * Gets the element length at closest (ceil) position from the nth percentile in a list
+ * @param {Array|Object} list
+ * @param {number} percentile Closest hundreth percentile
+ * @param {string | string[]} sortIteratee Lodash's iteratee or iteratee list
+ */
+export function getItemOfPercentileLength(list, percentile, sortIteratee = "length") {
+  return (
+    String(sortBy(list, sortIteratee)[Math.ceil((list.length - 1) * (percentile / 100))]).length * 10
+    // 10 is the root document font-size
+  );
+}
```

Comment:
Few thoughts on code style:
1. need to check for list length - return value of this function is undefined if it's empty;
2. avoid one-liners - they add unnecessary cognitive load. Split this algorithm in steps: sort, then take item, then compute result;
3. using hardcoded font size doesn't feel good to me. Ideally, there should be a font size of popup container, or at least - of `<body>` tag.

Also, maybe add some limit below which the function will just enable Select's built-in measure function, and use this function only if there are too much items? In this case, you can omit item 1

Reply:
1. Agreed
2. and 3. I was actually gonna suggest to move the font-size away from it and having a general `getItemOfPercentile`, because the font-size will depend on the context. This way the "one"-liner would be quite simple :
```js
if (isArray(list) && list.length > 0) {
  const itemIndex = Math.ceil((list.length - 1) * (percentile / 100));
  return sortBy(list, sortIteratee)[itemIndex];
}
```

> Also, maybe add some limit below which the function will just enable Select's built-in measure function

Nice one 👍, we can create an util function for the dropdownWidth value that sets it to `false` if the length is below a given threshold (e.g.: 500)